src/browser/new-browser.ts

```diff
@@ -221,36 +228,50 @@ export class NewBrowser extends Browser {
         return this._wdProcess.gridUrl;
     }
 
-    protected async _addExecutablePath(
+    protected async _applyLocalBrowserCapabilities(
         config: BrowserConfig,
         capabilities: WebdriverIO.Capabilities,
     ): Promise<WebdriverIO.Capabilities> {
-        const { getNormalizedBrowserName, installBrowser } = await import("../browser-installer");
-        const normalizedBrowserName = getNormalizedBrowserName(this._config.desiredCapabilities?.browserName);
+        const { installBrowser } = await import("../browser-installer");
+        const browserNameW3C = getNormalizedBrowserName(config.desiredCapabilities?.browserName);
 
-        if (!normalizedBrowserName) {
+        if (!browserNameW3C) {
             throw new Error(
                 [
-                    `Running auto local "${this._config.desiredCapabilities?.browserName}" is unsupported`,
+                    `Running auto local "${config.desiredCapabilities?.browserName}" is unsupported`,
                     `Supported browsers: "chrome", "firefox", "safari", "edge"`,
                 ].join("\n"),
             );
         }
 
-        const executablePath = await installBrowser(
-            normalizedBrowserName,
-            this._config.desiredCapabilities?.browserVersion,
-            { shouldInstallWebDriver: false, shouldInstallUbuntuPackages: true },
-        );
+        const executablePath = await installBrowser(browserNameW3C, config.desiredCapabilities?.browserVersion, {
+            shouldInstallWebDriver: false,
+            shouldInstallUbuntuPackages: true,
+        });
 
         if (executablePath) {
-            const browserNameLowerCase = config.desiredCapabilities?.browserName?.toLowerCase() as string;
-            const { capabilityName } = headlessBrowserOptions[browserNameLowerCase];
-            capabilities[capabilityName] ||= {};
-            capabilities[capabilityName]!.binary ||= executablePath;
+            const capabilitySettings = headlessBrowserOptions[browserNameW3C];
+
+            if (!capabilitySettings) {
+                return capabilities;
+            }
+
+            capabilities[capabilitySettings.capabilityName] ||= {};
+            capabilities[capabilitySettings.capabilityName]!.binary ||= executablePath;
         }
 
-        return capabilities;
+        const filteredCapabilities: WebdriverIO.Capabilities = Object.keys(capabilities)
+            .filter(capabilityName => {
+                const isW3CCapability = W3C_CAPABILITIES.includes(capabilityName);
+                const isVendorSpecificCapability = VENDOR_CAPABILITIES[browserNameW3C].includes(capabilityName);
+
+                return isW3CCapability || isVendorSpecificCapability;
+            })
+            .reduce((acc, capabilityName) => {
+                return set(acc, [capabilityName], capabilities[capabilityName as keyof WebdriverIO.Capabilities]);
+            }, {});
+
+        return filteredCapabilities;
```

Comment:
Remove all capabilities, which are not "w3c capabilities" or "vendor specific capabilities", as local webdriver can't recognize those and fail to launch browser